src/appleseed/renderer/kernel/intersection/embreescene.cpp

```diff
@@ -250,17 +221,20 @@ namespace
         const ShadingRay&   shading_ray,
         RTCRay&             embree_ray)
     {
-        const RayBuffer buffer(shading_ray.m_org, shading_ray.m_dir);
+        embree_ray.org_x = static_cast<float>(shading_ray.m_org.x);
```

Comment:
Can you explain this revert?

Reply:
You were casting and copying manually the org and dir from shading_ray to RayBuffer only to copy them again to the embree ray, this time using mm_store_ps.

You can copy them directly to the embree ray and avoid the store_ps intrinsics.

Reply:
Ah, okay. This was done based on embree's performance tips. They are in documentation pdf located somewhere in embree distribution folder in built appleseed deps